From c9cf9156754677a4064c506af215cb02165a03d3 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 27 Feb 2019 11:31:42 +0200
Subject: [PATCH 017/269] s32g: Update generic timer frequency

Initialize the generic timer with the correct frequency. One immediate
consequence of the old, incorrect value was that the U-Boot autoboot
timer would not decrement properly.

Issue: ALB-3702

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/s32g/include/platform_def.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 5a167124d..00d6c1eb8 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -49,7 +49,8 @@
 #define PLAT_MAX_PWR_LVL_STATES		2
 
 #define PLAT_PRIMARY_CPU		0x0
-#define COUNTER_FREQUENCY		0x40000000 /* FXOSC */
+/* Generic timer frequency; this goes directly into CNTFRQ_EL0 */
+#define COUNTER_FREQUENCY		0x00989680	/* 10MHz */
 
 /* GIC (re)definitions, some from U-Boot's asm/gic.h and the s32gen1 config */
 #define S32G275_GIC_BASE	0x50800000
-- 
2.17.1

